internal/archive/extractor_test.go

```diff
@@ -0,0 +1,224 @@
+			true,
+		},
+		{
+			"Zip: Returns no error if file is a valid archive",
+			validZip,
+			false,
+		},
+		{
+			"7z: Returns no error if file is a valid archive",
+			valid7z,
+			false,
+		},
+		{
+			"Rar: Returns no error if file is a valid archive",
+			validRar,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			if err := Extract(tt.source); (err != nil) != tt.wantErr {
+				t.Errorf("Extract(%q) error = %v, wantErr %v", tt.source, err, tt.wantErr)
+			}
+
+			// Assert file contents and cleanup
+			if !tt.wantErr {
+				followUpAssertAndCleanUp(t, extractionPath)
+			}
+		})
+	}
+}
+
+func ExampleExtractTo() {
+	err := ExtractTo("/path/to/archive.zip", "/destination/folder")
+	if err != nil {
+		fmt.Printf("Zip archive extraction failure: %v", err)
+	} else {
+		fmt.Println("Zip archive was successfully extracted")
+	}
+	// Output: Zip archive extraction failure: file /path/to/archive.zip not found
+}
+
+func TestExtractTo(t *testing.T) {
+	type args struct {
+		source      string
+		destination string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			"Returns error if file does not exist",
+			args{missingArchive, extractToPath},
+			true,
+		},
+		{
+			"Returns error if file is a directory",
+			args{existingFolder, extractToPath},
+			true,
+		},
+		{
+			"Returns error if file not a valid archive",
+			args{nonArchiveFile, extractToPath},
+			true,
+		},
+		{
+			"Zip: Returns error if file not a valid archive",
+			args{invalidZip, extractToPath},
+			true,
+		},
+		{
+			"7z: Returns error if file not a valid archive",
+			args{invalid7z, extractToPath},
+			true,
+		},
+		{
+			"Rar: Returns error if file not a valid archive",
+			args{invalidRar, extractToPath},
+			true,
+		},
+		{
+			"Returns error if provided path is not writable",
+			args{validZip, nonWritableDir},
+			true,
+		},
+		{
+			"Zip: Returns no error if file is a valid archive",
+			args{validZip, extractToPath},
+			false,
+		},
+		{
+			"Rar: Returns no error if file is a valid archive",
+			args{validRar, extractToPath},
+			false,
+		},
+		{
+			"Rar: Returns no error if file is a valid archive",
+			args{validRar, extractToPath},
+			false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			skipWindowsNonWritableDirScenario(t, tt.args.destination, tt.name)
+
+			if err := ExtractTo(tt.args.source, tt.args.destination); (err != nil) != tt.wantErr {
+				t.Errorf("Extract(%q, %q) error = %v, wantErr %v", tt.args.source, tt.args.destination, err, tt.wantErr)
+			}
+			// Assert file contents and cleanup
+			if !tt.wantErr {
+				followUpAssertAndCleanUp(t, tt.args.destination)
+			}
+		})
+	}
+}
+
+func skipWindowsNonWritableDirScenario(t *testing.T, destination string, scenarioName string) {
+	if destination == nonWritableDir && runtime.GOOS == "windows" {
+		t.Skipf("Skip %q test in windows", scenarioName)
+	}
+}
```

Comment:
It might be the disk volume is not `C` but another letter. I have no clue what is the letter in the GitHub CI, so I guess this code should work. I am getting the `pwd` that should include disk drive letter (e.g. `C:` or `F:` or whatever) and then join with with the directory (e.g. in case of `F:` that would be `F:/Windows`). 

```suggestion
func windowsDir(t *testing.T) string {
	path, err := os.Getwd()
	if err != nil {
		t.Fatalf("Cannot get CWD on Windows: %v", err)
	}

	return filepath.Join(filepath.VolumeName(path), filepath.Dir("/Windows"))
}
```